### PR TITLE
Add support for opencore ethernet MAC.

### PR DESCRIPTION
**Pull request by floitsch**
This MAC is supported in Qemu (but not in actual hardware).

### CODE REVIEW

File: src/resources/ethernet_esp32.cc [hunk 1]
```diff
@@ -200,7 +203,16 @@ PRIMITIVE(init_esp32) {
   mac_config.smi_mdio_gpio_num = mdio_num;
 
   // TODO(anders): If phy initialization fails, we're leaking this.
```
Comment (kasperl) [inline src/resources/ethernet_esp32.cc:205]: Maybe fix this while you're in here? Might be harder than you would think.

File: lib/net/modules/ethernet.toit [hunk 2]
```diff
@@ -39,11 +41,20 @@ class Ethernet:
 
     if mac_chip == MAC_CHIP_ESP32:
       resource_group_ = ethernet_init_esp32_
+        mac_chip
         phy_chip
         phy_addr
         (phy_reset ? phy_reset.num : -1)
         mac_mdc.num
         mac_mdio.num
+    else if mac_chip == MAC_CHIP_OPENETH:
```
Comment (kasperl) [inline lib/net/modules/ethernet.toit:50]: Maybe share more of this code with the case above?

**Review by kasperl — APPROVED** [review]
Looks good to me.

How much extra space does this take up?

File: lib/net/modules/ethernet.toit (hunk 2 above)
Comment (floitsch) [reply lib/net/modules/ethernet.toit]: done.

File: src/resources/ethernet_esp32.cc (hunk 1 above)
Comment (floitsch) [reply src/resources/ethernet_esp32.cc:205]: Will do it in a separate PR.

**Pull request merged**